### PR TITLE
Remove pdftk

### PR DESCRIPTION
**Pull request by clareconstantine**
The pdftk library is causing issues when creating adhocs. We use this library to add page numbers to pdfs, and we have decided that we are ok not adding page numbers to any pdfs we generate in the future, so I'm removing the library.

## Links

- jira ticket: [PLAT-1408](https://codedotorg.atlassian.net/browse/PLAT-1408)

## Testing story

I tested locally that the remaining pdf tests pass after uninstalling pdftk.

## Follow-up work

I think we may be able to remove `enscript` and `ps2pdf`, and I have a [Jira ticket](https://codedotorg.atlassian.net/browse/PLAT-1585) to investigate that.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by Hamms — APPROVED** [review]
I believe the removed `number_pdf` method is also our only usage of `enscript`, so we can also remove that as a dependency!

It might even be cleanest to do that in this PR, but either way this change LGTM; thank you!!

**Comment by Hamms** [comment]
Oh lol, I see you already noted the enscript possibility in the Follow-up section! Makes sense to me to do that as a separate PR given that there might be even more cleanup possible

File: pegasus/sites/virtual/collate_pdfs.rake [hunk 1]
```diff
@@ -14,20 +14,13 @@ def collate_to_pdf_to_fetch_file(collate_file)
   fetchfile_path = "#{v3_path}.fetch"
   file fetchfile_path => ([collate_file] + source_paths) do
     # Convert all local PDF paths to URLs (since some may be non-local .fetch files)
-    options, pdf_paths = PDF.parse_collate_file(collate_file)
+    _, pdf_paths = PDF.parse_collate_file(collate_file)
```
Comment (davidsbailey) [inline pegasus/sites/virtual/collate_pdfs.rake:17]: it looks like the first return value of parse_collate_file is now not used anywhere in our codebase, so if you want to go 1 level deeper you could change `parse_collate_file` to only return `all_paths`: https://github.com/code-dot-org/code-dot-org/blob/87854385833d6cb35d2bd5071d07f0dd32b478ae/lib/pdf/collate.rb#L41

File: pegasus/test/test_pdf_merger.rb [hunk 2]
```diff
@@ -69,8 +63,7 @@ def assert_local_dependencies
         # Require dependencies on linux which our servers use
         assert found, missing_dependency_message
       else
-        # Skip these tests if we are on OSX and missing dependencies;
-        # There's a known problem installing pdftk on OSX right now (see SETUP.md)
+        # Skip these tests if we are on OSX and missing dependencies
```
Comment (davidsbailey) [inline pegasus/test/test_pdf_merger.rb:66]: nice pattern above for explicitly checking for missing dependencies 😎

**Review by davidsbailey — APPROVED** [review]
Awesome work Clare, thanks for doing this! there's definitely more of a tangle around pdftk than I would have guessed, especially when you consider all the logic and docs to support the workarounds.

**Comment by davidsbailey** [comment]
@clareconstantine I am wondering if we can also get rid of any of these? https://github.com/code-dot-org/code-dot-org/blob/e84ab69c51a5d3ca55f4b0bb3c789ffb6e8f0c11/Gemfile#L111-L113

**Pull request closed without merging**